src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -339,14 +340,21 @@ private void doRetrieve(SCMHeadObserver observer, TaskListener listener, GHRepos
                 return 0; // TODO
             }
             @Override public boolean exists(@Nonnull String path) throws IOException {
-                try {
-                    repo.getFileContent(path, ref);
-                    listener.getLogger().format("      %s exists in this %s%n", path, thing);
-                    return true;
-                } catch (FileNotFoundException x) {
-                    listener.getLogger().format("      %s does not exist in this %s%n", path, thing);
-                    return false;
+                List<GHContent> directoryContent = repo.getDirectoryContent("/", ref);
+                for (GHContent content : directoryContent) {
+                    if (content.isFile()) {
+                        if (content.getName().equals(path)) {
+                            listener.getLogger().format("      ‘%s’ exists in this %s%n", path, thing);
+                            return true;
+                        }
+                        if (content.getName().equalsIgnoreCase(path)) {
+                            listener.getLogger().format("      ‘%s’ not found (but found ‘%s’, search is case sensitive) in this %s, skipping%n", path, content.getName(), thing);
+                            return false;
+                        }
+                    }
                 }
+                listener.getLogger().format("      %s does not exist in this %s%n", path, thing);
```

Comment:
Might want to put quotes around the pathname here for consistency with the other two messages.